### PR TITLE
[ReorderableListView] remove extra margin added after picking up the item

### LINKED ISSUE #58411: ReorderableListView listitems jump down on dragging
## Steps to Reproduce

See the behaviour in this video: https://www.youtube.com/watch?v=BUxca5SsI1Q
As you can see, the moment item#5 is dragging, all the items below it jump a few pixels down.

**Expected results:** 
When starting to drag an item, the other items should remain in place.

**Actual results:** 
The items below the selected item jump down a few pixels.

<details>
  <summary>Logs</summary>

```
```

```
```

```
[✓] Flutter (Channel stable, v1.17.1, on Mac OS X 10.15.5 19F96, locale nl-NL)
    • Flutter version 1.17.1 at /Users/robin/development/flutter
    • Framework revision f7a6a7906b (3 weeks ago), 2020-05-12 18:39:00 -0700
    • Engine revision 6bc433c6b6
    • Dart version 2.8.2

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.3)
    • Android SDK at /Users/robin/Library/Android/sdk
    • Platform android-29, build-tools 29.0.3
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.5)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.5, Build version 11E608c
    • CocoaPods version 1.9.2

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.8052
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] Connected device (1 available)
    • Android SDK built for x86 64 • emulator-5554 • android-x64 • Android 10 (API 29) (emulator)

• No issues found!

```

</details>

### PR DESCRIPTION
**Pull request by nero-angela**
## Description
As you can see under the gif when picking up an item in `ReorderableListView`, the others below the item move slightly downward. this is because the extra margin is added to the selected item after lifting. So I removed the extra margin and modified the related test.

|before|after|
|:--:|:--:|
|before|after|

### Golden test image
|reorderable_list_test.vertical.drop_area.png|reorderable_list_test.horizontal.drop_area.png|
|:--:|:--:|
|reorderable_list_test vertical drop_area|reorderable_list_test horizontal drop_area|

## Related Issues
Fixes https://github.com/flutter/flutter/issues/58411

## Tests

I modified the following tests: 
- [properly determines the vertical drop area extents](https://github.com/flutter/flutter/blob/780f5f99044ce7fcbcd868e2cf6878da02022db5/packages/flutter/test/material/reorderable_list_test.dart#L144)
- [properly determines the horizontal drop area extents](https://github.com/flutter/flutter/blob/780f5f99044ce7fcbcd868e2cf6878da02022db5/packages/flutter/test/material/reorderable_list_test.dart#L659)

## Checklist

Before you create this PR, confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/material/reorderable_list.dart [hunk 1]
```diff
@@ -207,7 +207,7 @@ class _ReorderableListContentState extends State<_ReorderableListContent> with T
   static const double _defaultDropAreaExtent = 100.0;
 
   // The additional margin to place around a computed drop area.
-  static const double _dropAreaMargin = 8.0;
+  static const double _dropAreaMargin = 0.0;
```
Comment (dkwingsmt) [inline packages/flutter/lib/src/material/reorderable_list.dart:210]: This variable should no longer be necessary. Remove to simplify code.

File: packages/flutter/test/material/reorderable_list_test.dart [hunk 2]
```diff
@@ -178,8 +178,8 @@ void main() {
         }
 
         const double kNonDraggingListHeight = 292.0;
-        // The list view pads the drop area by 8dp.
-        const double kDraggingListHeight = 300.0;
+        // The list view pads the drop area by 0dp.
+        const double kDraggingListHeight = 292.0;
```
Comment (dkwingsmt) [inline packages/flutter/test/material/reorderable_list_test.dart:182]: Combine `kNonDraggingListHeight` and `kDraggingListHeight` to simplify code.

**Review by dkwingsmt — CHANGES_REQUESTED** [review]
Good change, with small change requests.

Also it'd be better if you can add a golden test to make sure it looks exactly how we want after picking up the item. If you're up for the challenge, refer to https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter or search for `matchesGoldenFile` in existing test files.

**Comment by nero-angela** [comment]
Thank you @dkwingsmt ! I've simplified the unnecessary code and added golden test. The images are in the description. :)
This is my first time making golden file changes, I'm not sure if I performed the golden test correctly 🤔
Could you please take a look?

**Comment by dkwingsmt** [comment]
The Linux analyzer seems unhappy. Can you resolve the lint errors?

**Comment by nero-angela** [comment]
sorry for missing declaring const. @dkwingsmt I've resolved the lint error.

**Review by dkwingsmt — APPROVED** [review]
LGTM

**Comment by dkwingsmt** [comment]
Theoretically the bot should comment that a golden test has been modified... I'll check it out next week.

**Comment by nero-angela** [comment]
Okay. Thank you for reviewing. Check it out and tell me if there is anything I need to do.
Have a nice weekend :)

**Comment by justinmc** [comment]
@dkwingsmt @nero-angela The golden tests added in this PR failed when I tried to revert a TextField layout PR from just before this: https://github.com/flutter/flutter/pull/66027

Can you guys think of any reason why that would happen?  These new tests have nothing to do with TextField or InputDecoration layout, right?

**Comment by justinmc** [comment]
Actually it seems to be happening on many PRs:
https://github.com/flutter/flutter/pull/61981
https://github.com/flutter/flutter/pull/65884

**Comment by Piinks** [comment]
These images don't appear to have arrived on the gold dashboard after landing: https://flutter-gold.skia.org/
There is nothing under by blame, and I can't find reference to the tests. Checking the logs now on master.
fyi @kjlubick

**Comment by Piinks** [comment]
Looks like our luci recipes have been changed and disabled gold again. Will file an infra ticket.

**Comment by Piinks** [comment]
https://github.com/flutter/flutter/issues/64969 for tracking

**Pull request merged**